modal-go/auth_token_manager.go

```diff
@@ -51,33 +57,65 @@ func NewAuthTokenManager(client pb.ModalClientClient, logger *slog.Logger) *Auth
 // Start the token refresh goroutine.
 // Returns an error if the initial token fetch fails.
 func (m *AuthTokenManager) Start(ctx context.Context) error {
+	m.mu.Lock()
+	if m.running {
+		m.mu.Unlock()
+		return nil
+	}
+	m.running = true
 	refreshCtx, cancel := context.WithCancel(ctx)
 	m.cancelFn = cancel
-	if _, err := m.FetchToken(refreshCtx); err != nil {
-		cancel()
-		m.cancelFn = nil
+	m.mu.Unlock()
+
+	if err := m.runFetch(refreshCtx); err != nil {
+		m.Stop()
 		return fmt.Errorf("failed to fetch initial auth token: %w", err)
 	}
+
 	go m.backgroundRefresh(refreshCtx)
 	return nil
 }
 
 // Stop the refresh goroutine.
 func (m *AuthTokenManager) Stop() {
-	if m.cancelFn != nil {
-		m.cancelFn()
-		m.cancelFn = nil
+	m.mu.Lock()
+	m.running = false
+	cancelFn := m.cancelFn
+	m.cancelFn = nil
+	m.mu.Unlock()
+
+	if cancelFn != nil {
+		cancelFn()
 	}
 }
 
-// GetToken returns the current cached token.
-func (m *AuthTokenManager) GetToken(ctx context.Context) (string, error) {
-	token := m.GetCurrentToken()
+// runFetch fetches a token, ensuring only one fetch is in progress at a time.
+func (m *AuthTokenManager) runFetch(ctx context.Context) error {
+	_, err, _ := m.fetchGroup.Do("fetch", func() (interface{}, error) {
+		return m.FetchToken(ctx)
+	})
+	return err
+}
 
-	if token != "" && !m.IsExpired() {
+// GetToken returns a valid auth token.
+// If the current token is expired and the manager is running, triggers an on-demand refresh.
+func (m *AuthTokenManager) GetToken(ctx context.Context) (string, error) {
+	if token := m.GetCurrentToken(); token != "" && !m.IsExpired() {
 		return token, nil
 	}
 
+	m.mu.Lock()
+	running := m.running
+	m.mu.Unlock()
+
+	if running {
```

Comment:
Is the `running` flag check needed here? Guess you can `GetToken` even if it's not `running` -- unless you intend to cache the value?